core/src/main/java/io/grpc/internal/PickFirstLeafLoadBalancer.java

```diff
@@ -468,6 +474,13 @@ private final class HealthListener implements SubchannelStateListener {
 
     @Override
     public void onSubchannelState(ConnectivityStateInfo newState) {
+      if (notAPetiolePolicy) {
+        log.log(Level.WARNING,
+            "Ignoring health status {0} for subchannel {1} as this is not a petiole policy",
```

Comment:
should we say "not a petiole policy" or "there is no petiole policy". This policy is always leaf policy, the difference is whether there is a petiole policy or this is top level policy. Health check can not be used with health producer + pick first (without petiole policy).

A61:
>>This set of policies, which we will refer to as "[petiole](https://en.wikipedia.org/wiki/Petiole_(botany))" policies, includes the following:
round_robin (see [gRPC Load Balancing](https://github.com/grpc/grpc/blob/master/doc/load-balancing.md#round_robin))
weighted_round_robin (see [gRFC A58](https://github.com/grpc/proposal/blob/master/A58-client-side-weighted-round-robin-lb-policy.md))
ring_hash (see [gRFC A42](https://github.com/grpc/proposal/blob/master/A42-xds-ring-hash-lb-policy.md))
least_request (see [gRFC A48](https://github.com/grpc/proposal/blob/master/A48-xds-least-request-lb-policy.md) -- currently supported in Java and Go only)

Reply:
Updated message to say "not under a petiole policy"